src/subclass/input_stream.rs

```diff
@@ -295,6 +301,58 @@ mod tests {
         }
     }
 
+    impl SeekableImpl for SimpleInputStream {
+        fn tell(&self, _seekable: &Seekable) -> i64 {
+            *self.pos.borrow() as i64
+        }
+
+        fn can_seek(&self, _seekable: &Seekable) -> bool {
+            true
+        }
+
+        fn seek(
+            &self,
+            _seekable: &Seekable,
+            offset: i64,
+            type_: glib::SeekType,
+            _cancellable: Option<&Cancellable>,
+        ) -> Result<(), glib::Error> {
+            let mut pos = self.pos.borrow_mut();
+            match type_ {
+                glib::SeekType::Set => {
+                    *pos = offset as usize;
+                    Ok(())
+                }
+                glib::SeekType::Cur => {
+                    if offset < 0 {
+                        *pos -= (-offset) as usize;
```

Comment:
Shouldn't we check if offset is superior than pos in here?

Reply:
It's in the test, and this will panic if it fails (overflows/etc panic in debug/test builds)

Reply:
You know I'm not a big fan of panic. :p

Shouldn't we just use `max(pos, (-offset) as usize)`?

Reply:
Well, this is a test. It's supposed to panic if things fail, just like all the `assert!` later in the test. Do you want me to make this more explicit by adding assertions here?

In a real implementation you would return a `glib::Error` here.

Reply:
Oh crap! I thought it was still "normal" code and not test. Nevermind and sorry for the noise...